seqmagick/subcommands/convert.py

```diff
@@ -315,7 +307,13 @@ def transform_file(source_file, destination_file, arguments):
         # loading the entire sequence file up into memory.
         logging.info("Applying transformations, writing to %s",
                 destination_file)
-        SeqIO.write(records, destination_file, destination_file_type)
+        # FIXME: to put annotations, all records are loaded in memory, which
```

Comment:
Hm, I'm wondering if you could make a wrapper iterator object here. Something like
```
def append_annotation_iterator(records_iterator):
    for record in records_iterator:
        record.annotations["molecule_type"] = "DNA"
        yield record

annotated_records = append_annotation_iterator(records)
SeqIO.write(annotated_records, destination_file, destination_file_type)
```

Reply:
Just a suggestion but I might be over-looking something. At leas this approach would avoid having to load everything into memory which may be a more serious problem.

Reply:
In fact, this seems to pass all tests when done this way. Could you give it a try on your end?

Reply:
Hi Jared, this is confirmed working on my end as well.  I'll push this now.  Thanks you, I believe I learned something cool today.  :)

Reply:
Awesome, thanks Étienne! this all looks really good I think we're pretty much there. Can you confirm (help me understand) the `record.annotations["molecule_type"]` Should always be equal to `"DNA"`? I'm not sure I understand the context here :/